examples/example-app-router/src/components/LocaleSwitcherSelect.tsx

```diff
@@ -18,11 +19,21 @@ export default function LocaleSwitcherSelect({
   const router = useRouter();
   const [isPending, startTransition] = useTransition();
   const pathname = usePathname();
+  const params = useParams();
 
   function onSelectChange(event: ChangeEvent<HTMLSelectElement>) {
     const nextLocale = event.target.value;
     startTransition(() => {
-      router.replace(pathname, {locale: nextLocale});
+      router.replace(
+        {
+          pathname,
+          // TypeScript validates that only known `params` are used in combination
+          // with a given `pathname`. Since the two will always match for the current
+          // route, we can skip runtime checks.
+          params: params as any
```

Comment:
The situation is slightly different here: Since there are no dynamic params in this example, there can never be any `params`. However, I'd agree that it's beneficial if we show a sample that works for either case.

This would work:

```suggestion
          // @ts-expect-error -- TypeScript validates that only known `params` are used in combination
          // with a given `pathname`. Since the two will always match for the current
          // route, we can skip runtime checks.
          params
```

That would work in the example in the docs too. We could update the docs accordingly so these are in sync.

What do you think?

Reply:
Oh ok, I understand. Whatever you want also works for me. I had just stumbled onto this issue because I started a project using example-app-router as my starting point and eventually created a dynamic route causing the error.

Your suggested placement of the `@ts-expect-error` did not dismiss the error. I actually had to place the `@ts-expect-error` comment above the entire `{ pathname, params }` object like this:
```
startTransition(() => {
  router.replace(
    // @ts-expect-error -- TypeScript will validate that only known `params`
    // are used in combination with a given `pathname`. Since the two will
    // always match for the current route, we can skip runtime checks.
    { pathname, params },
    { locale: nextLocale },
  );
});
```
Does this work for you?

Also, when you say "We could update the docs accordingly so these are in sync", are you saying the docs at `docs/pages/docs/routing/navigation.mdx` should include the `@ts-expect-error` comment as well?

Reply:
Sure, that looks good! Yep, would you mind updating `navigation.mdx` to match the updated implementation in `example-app-router`?